src/components/TaskForm.tsx

```diff
@@ -0,0 +1,46 @@
+// src/components/TaskForm.tsx
+import React, { useState } from 'react';
+import "../styles/TaskForm.css"
+interface TaskFormProps {
+    onSubmit: (text: string, category: string, priority: 'High' | 'Medium' | 'Low') => void;
+    initialValues?: { text: string; category: string; priority: 'High' | 'Medium' | 'Low' };
+}
+
+const TaskForm: React.FC<TaskFormProps> = ({ onSubmit, initialValues }) => {
+    const [text, setText] = useState(initialValues?.text || '');
+    const [category, setCategory] = useState(initialValues?.category || '');
+    const [priority, setPriority] = useState<'High' | 'Medium' | 'Low'>(initialValues?.priority || 'Low');
+
+    const handleSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+        onSubmit(text, category, priority);
+        setText('');
+        setCategory('');
+        setPriority('Low');
+    };
+
+    return (
+        <form onSubmit={handleSubmit} className="task-form">
```

Comment:
Since we are using a form, have use considered using uncontrolled input component?

Reply:
updated